pallets/creditcoin/src/migrations/v8.rs

```diff
@@ -0,0 +1,232 @@
+use super::{vec, Vec};
+use super::{AccountIdOf, BlockNumberOf, HashOf, Migrate, MomentOf, PhantomData};
+
+use crate::types::CollectedCoinsStruct;
+
+use super::v6::Task as OldTask;
+use crate::{AddressId, Config, TaskId, UnverifiedCollectedCoins, UnverifiedTransfer};
+use crate::{CollectedCoinsId, ExternalTxId};
+use frame_support::storage_alias;
+use frame_support::weights::Weight;
+use frame_support::{pallet_prelude::*, traits::Get};
+use parity_scale_codec::{Decode, Encode};
+
+#[derive(Clone, Encode, Decode)]
+pub struct OldCollectedCoinsStruct<Hash, Balance> {
+	pub to: AddressId<Hash>,
+	pub amount: Balance,
+	pub tx_id: ExternalTxId,
+}
+
+pub(super) struct Migration<Runtime>(PhantomData<Runtime>);
+
+impl<Runtime: Config> Migration<Runtime> {
+	pub(super) fn new() -> Self {
+		Self(PhantomData)
+	}
+}
+
+#[storage_alias]
+pub type PendingTasks<T: Config> = StorageDoubleMap<
+	TaskScheduler,
+	Identity,
+	BlockNumberOf<T>,
+	Identity,
+	TaskId<HashOf<T>>,
+	OldTask<AccountIdOf<T>, BlockNumberOf<T>, HashOf<T>, MomentOf<T>>,
+>;
+
+mod new {
+
+	use crate::Task;
+
+	use super::*;
+
+	#[storage_alias]
+	pub type PendingTasks<T: Config> = StorageDoubleMap<
+		TaskScheduler, // the prefix for the storage item, which is generally the name of the pallet that defines the storage. We use an identifier instead of a string here because that's what the macro expects.
+		Identity,
+		BlockNumberOf<T>,
+		Identity,
+		TaskId<HashOf<T>>,
+		Task<AccountIdOf<T>, BlockNumberOf<T>, HashOf<T>, MomentOf<T>>, // the `Task` we copied into this migration
+	>;
+}
+
+impl<T: Config> Migrate for Migration<T> {
+	fn pre_upgrade(&self) -> Vec<u8> {
+		vec![]
+	}
+
+	fn post_upgrade(&self, _blob: Vec<u8>) {
+		assert_eq!(
+			StorageVersion::get::<crate::Pallet<T>>(),
+			8,
+			"expected storage version to be 8 after migrations complete"
+		);
+	}
+	fn migrate(&self) -> Weight {
+		let mut weight: Weight = Weight::zero();
+
+		let weight_each = T::DbWeight::get().reads_writes(1, 1);
+
+		crate::CollectedCoins::<T>::translate::<OldCollectedCoinsStruct<T::Hash, T::Balance>, _>(
+			|_k: CollectedCoinsId<T::Hash>, y: OldCollectedCoinsStruct<T::Hash, T::Balance>| {
+				weight = weight.saturating_add(weight_each);
+
+				Some(CollectedCoinsStruct {
+					to: y.to,
+					amount: y.amount,
+					tx_id: y.tx_id,
+					contract_type: crate::types::ContractType::GCRE,
+				})
+			},
+		);
+
+		new::PendingTasks::<T>::translate(
+			|_: BlockNumberOf<T>,
+			 _: TaskId<HashOf<T>>,
+			 z: OldTask<AccountIdOf<T>, BlockNumberOf<T>, HashOf<T>, MomentOf<T>>| {
+				weight = weight.saturating_add(weight_each);
+
+				match z {
+					OldTask::VerifyTransfer(pending) => {
+						let new = UnverifiedTransfer {
+							transfer: pending.transfer,
+							from_external: pending.from_external,
+							to_external: pending.to_external,
+							deadline: pending.deadline,
+						};
+
+						return Some(crate::types::Task::VerifyTransfer(new));
```

Comment:
```suggestion
						Some(crate::types::Task::VerifyTransfer(new))
```

Clippy catches this locally, but it looks like CI hasn't run on this PR since your latest commits? Not sure what's up with that